ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/SandboxResetService.scala

```diff
@@ -69,14 +69,14 @@ class SandboxResetService(
       throw new StatusRuntimeException(
         Status.FAILED_PRECONDITION.withDescription("Sandbox server is currently being resetted"))
 
-    val servicesAreDown = Promise[Unit]()
-    // We need to run this asynchronously since otherwise we have a deadlock: `buildAndStartServer` will block
-    // until all the in flight requests have been served, so we need to schedule this in another thread so that
-    // the code that clears the in flight request is not in an in flight request itself.
-    getEc().execute({ () =>
-      logger.info(s"Stopping and starting the server.")
-      servicesAreDown.completeWith(resetAndRestartServer())
-    })
-    servicesAreDown.future
+    logger.info(s"Stopping and starting the server.")
+
+    // We need to run this asynchronously since otherwise we have a deadlock: `buildAndStartServer`
+    // will block until all the in flight requests have been served, so we need to schedule this in
+    // another thread so that the code that clears the in flight request is not in an in flight
+    // request itself.
+    resetAndRestartServer()
+
+    Future.successful(())
```

Comment:
Previously we returned a promise that was completed by `resetAndRestartServer()`. Now we basically always signal that reset is over. Was this intentional?

Reply:
Yes. Rather than the Sandbox code having to run the reset operation asynchronously, I moved the behavior to the SandboxResetService in https://github.com/digital-asset/daml/pull/4802/commits/9bacd95e56236f9932ce4f5232a8bb86c6a4758d.

We can do this because in Sandbox Classic™, we _actually_ wait until the API services have been closed, and in Sandbox Next, we wait until everything has been `.release()`d (and also check the API services have been closed).